service/src/main/java/bio/terra/pipelines/common/utils/PipelineVariableTypesEnum.java

```diff
@@ -23,9 +24,14 @@ public <T> T cast(String fieldName, Object value, TypeReference<T> typeReference
     @Override
     public String validate(PipelineInputDefinition pipelineInputDefinition, Object value) {
       String fieldName = pipelineInputDefinition.getName();
-      if (cast(fieldName, value, new TypeReference<String>() {}) == null) {
+      String castValue = cast(fieldName, value, new TypeReference<String>() {});
+      if (castValue == null) {
         return "%s must be a string".formatted(fieldName);
       }
+      if (!VALID_STRING_PATTERN.matcher(castValue).matches()) {
```

Comment:
i know your ticket talked about trimming before validating for spaces.  Does trimming already happen before this validation or is that something we should do here?

Reply:
`FILE` reuses the `cast` method from `STRING` (see [here](https://github.com/DataBiosphere/terra-scientific-pipelines-service/blob/main/service/src/main/java/bio/terra/pipelines/common/utils/PipelineVariableTypesEnum.java#L110-L126)), so it gets the same trimming done. there are also a few cases for it in both the `FILE` and `FILE_ARRAY` cast+validation tests